### PR TITLE
App crud logic

### PR DESCRIPTION
**Pull request by Ikilonchic**
Added logic for loading, deleting and creating apps)

### CODE REVIEW

File: src/services/SocialsApiService.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  static async post(authToken, { name }) {
+    const url = SocialsApiService.#buildUrl();
+
+    return axios.post(url, {
+      headers: {
+        Authentication: authToken,
+      },
+      data: {
+        name,
+      },
```
Comment (matroskin062) [inline src/services/SocialsApiService.js]: This code not works correct, it's not send "name" field.
You can do like this and it should work fine
```suggestion
     return axios.post(
      url,
      { name },
      {
        headers: {
          Authentication: authToken,
        },
      }
```

File: src/services/AppsApiService.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  static async post(authToken, { name }) {
+    const url = AppsApiService.#buildUrl();
+
+    return axios.post(url, {
+      headers: {
+        Authentication: authToken,
+      },
+      data: {
+        name,
+      },
```
Comment (matroskin062) [inline src/services/AppsApiService.js]: This code not works correct, it's not send "name" field.
You can do like this and it should work fine
```suggestion
    return axios.post(
      url,
      { name },
      {
        headers: {
          Authentication: authToken,
        },
      }
```

File: src/services/AppsApiService.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    const url = AppsApiService.#buildUrl();
+
+    return axios.get(url, {
+      headers: {
+        Authentication: authToken,
+      },
+    });
+  }
+
+  static async getById(authToken, { id }) {
```
Comment (matroskin062) [inline src/services/AppsApiService.js]: I think we don't need `async` keyword here and in other places

File: src/services/AppsApiService.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  static async updateById(authToken, { id, name }) {
+    const url = AppsApiService.#buildUrl();
+
+    return axios.put(url.href + `/${id}`, {
+      headers: {
+        Authentication: authToken,
+      },
+      data: {
+        name,
+      },
```
Comment (matroskin062) [inline src/services/AppsApiService.js]: here same problem as i described in `post` methods

**Review by matroskin062 — CHANGES_REQUESTED** [review]

**Comment by matroskin062** [comment]
Also u can define `headers` options on axios instance and it will make each request with this options
[Axios instacance docs](https://github.com/axios/axios#creating-an-instance)

File: src/components/AppInfo/AppInfo.jsx [hunk 5]
```diff
@@ -16,20 +17,44 @@ import {
 } from 'react-bootstrap';
 
 import styles from './AppInfo.module.scss';
+import selector from './AppInfo.selector';
 
 import appIcon from '../../assets/app-icon.svg';
 import googleIcon from '../../assets/google-avatar.png';
 import facebookIcon from '../../assets/facebook-avatar.png';
+import { useEffect } from 'react';
```
Comment (alexgul1) [inline src/components/AppInfo/AppInfo.jsx]: n this file there is already an import from the `react` library, this import needs to be moved above

File: src/components/AppInfo/AppInfo.jsx [hunk 6]
```diff
@@ -39,17 +64,20 @@ const AppInfo = () => {
...
               <Col>
-                {/*TODO get from backend basic info*/}
                 <FormGroup>
                   <FormLabel>Name</FormLabel>
-                  <FormControl type={'text'} />
+                  <FormControl
+                    type={'text'}
+                    readOnly
+                    defaultValue={current && current.name}
+                  />
```
Comment (alexgul1) [inline src/components/AppInfo/AppInfo.jsx]: why name field is read only? we can change this value, only appId is unique identificator

File: src/components/CreateAppModal/CreateAppModal.jsx [hunk 7]
```diff
@@ -18,37 +36,67 @@ const CreateAppModal = () => {
...
-          </Button>
-        </Modal.Footer>
+        <Formik
+          initialValues={initialValues}
+          validate={validate}
+          onSubmit={onSubmit}
+        >
+          {({
+            values,
+            errors,
```
Comment (alexgul1) [inline src/components/CreateAppModal/CreateAppModal.jsx]: add `yup` library for validation)

File: src/services/AppsApiService.js (hunk 3 above)
Comment (alexgul1) [reply src/services/AppsApiService.js]: > I think we don't need `async` keyword here and in other places

agree with Dima, its not good block stream

**Review by alexgul1 — CHANGES_REQUESTED** [review]
`src/ducks/apps.js` - maybe better to make a separation for working with all applications and one specific

**Comment by alexgul1** [comment]
also need to cover the code with tests

**Review by alexgul1 — APPROVED** [review]

**Review by matroskin062 — APPROVED** [review]

**Pull request merged**